compiler/x/amd64/codegen/OMRMemoryReference.cpp

```diff
@@ -142,27 +142,20 @@ void OMR::X86::AMD64::MemoryReference::finishInitialization(
    // Figure out whether we need to allocate a register for the address
    //
    bool mightNeedAddressRegister;
-   if (!self()->getBaseRegister() &&
-       !self()->getIndexRegister() &&
-       (cg->needRelocationsForStatics() ||
-        cg->needClassAndMethodPointerRelocations() ||
-        cg->needRelocationsForBodyInfoData() ||
-        cg->needRelocationsForPersistentInfoData()))
-      {
-      mightNeedAddressRegister = true;
-      }
-   else if (sr.getSymbol() != NULL && sr.isUnresolved())
-      {
-      // Once resolved, the address could be anything, so be conservative.
-      //
-      mightNeedAddressRegister = true;
-      }
-   else if (self()->getDataSnippet())
+   if (self()->getDataSnippet())
       {
       // Assume snippets are in RIP range
       //
       mightNeedAddressRegister = false;
       }
+   else if (!self()->getBaseRegister() && !self()->getIndexRegister() &&
```

Comment:
can we put the getIndexRegister on its own line since it will make reading this a lot easier

Reply:
Sure, it has been updated.